### ISSUE TITLE
`str expand` unexpectedly fails with nested braces, if the outermost defines only one item

### ISSUE THREAD

**Issue by goth-turtle** [nushell/nushell#14290]
### Describe the bug

Running this example:
```nu
"hello {{w,W}orld}" | str expand
```

Produces this error message:
```
Error:   × Expected Text
   ╭─[entry #15:1:1]
 1 │ "hello {{w,W}orld}" | str expand
   · ─────────┬─────────
   ·          ╰── Expected text at 7.
   ╰────
  help: Texts are only allowed before opening brace (`{`), after closing brace (`}`), or inside `{}`. Please take a look at the examples.
```

Some other combinations of braces that also fail:
- `"hello {world{,!}}" | str expand`
- `"{hello {w,W}orld}" | str expand`

The reverse does give the expected result though:
```
❯ "hello {world,World{!}}" | str expand
╭───┬──────────────╮
│ 0 │ hello world  │
│ 1 │ hello World! │
╰───┴──────────────╯
```

This also doesn't show up with `glob`:
```
❯ cd /tmp
❯ touch hello-world
❯ touch hello-World
❯ glob "hello-{{w,W}orld}"
╭───┬──────────────────╮
│ 0 │ /tmp/hello-world │
│ 1 │ /tmp/hello-World │
╰───┴──────────────────╯
```

### How to reproduce

1. Run the example above in nushell

### Expected behavior

I expected nu to evaluate `"hello {{w,W}orld}" | str expand` to `["hello world" "hello World"]`, or give a more descriptive error.

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.96.1                                                            |
| major              | 0                                                                 |
| minor              | 96                                                                |
| patch              | 1                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.79.0 (129f3b996 2024-06-10) (built from a source tarball) |
| cargo_version      | cargo 1.79.0                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, system-clipboard, trash                          |
| installed_plugins  |                                                                   |

The same error also shows up on main:

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.98.1                                                            |
| major              | 0                                                                 |
| minor              | 98                                                                |
| patch              | 1                                                                 |
| branch             | main                                                              |
| commit_hash        | addf8029316dea698af7d34d39f48dcf09223ae0                          |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.80.1 (3f5fd8dd4 2024-08-06) (built from a source tarball) |
| cargo_version      | cargo 1.80.0 (376290515 2024-07-16)                               |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | debug                                                             |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, trash                                            |
| installed_plugins  |                                                                   |

**Comment by fdncred** [comment]
`str expand` uses a crate called bracoxide to do this type of expansion. I'm wondering if this is an upstream problem rather than a nushell issue.

`glob` uses an entirely different crate to do globbing, called wax. I'm not surprised it works a little differently.

There's also a plugin called nu_plugin_bexpand that handles this syntax better and uses something other than bracoxide, which says to me that bracoxide may be at fault here. https://github.com/atahabaki/bracoxide

@atahabaki - Thoughts on this?

**Comment by atahabaki** [comment]
This issue is a bit weird, i'll look into it. Got my eyes wide open. I'll open a PR as soon as my fixes complete.

**Comment by atahabaki** [comment]
When I first designed this library, I planned to throw an error for single-item brace expressions. But later on, I decided I wanted my crate to be as user-friendly as possible, so I scrapped that idea. I honestly never expected anyone would use nested single-item brace expressions. This moment is definitely one I'll remember for the rest of my life!

**Comment by atahabaki** [comment]
Fixed, v1.0.4 out, soon I'all add a PR.
